test/wallet-events-test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const assert = require('bsert');
+const Address = require('../lib/primitives/address');
+const FullNode = require('../lib/node/fullnode');
+
+describe('WalletDB', function () {
+  const node = new FullNode({
+    memory: true,
+    network: 'regtest',
+    plugins: [require('../lib/wallet/plugin')]
+  });
+
+  const { wdb } = node.require('walletdb');
+  let wallet;
+
+  async function mineBlocks(n, addr) {
+    addr = addr ? addr : new Address().toString('regtest');
+    for (let i = 0; i < n; i++) {
+      const block = await node.miner.mineBlock(null, addr);
+      await node.chain.add(block);
+    }
+  }
+
+  before(async () => {
+    await node.open();
+    wallet = await wdb.get('primary');
+  });
+
+  after(async () => {
+    await node.close();
+  });
+
+  it('should emit `tx` events', async () => {
+    const waiter = new Promise((resolve) => {
+      wdb.once('tx', (w, tx) => resolve([w, tx]));
+    });
+
+    const walletReceive = await wallet.receiveAddress();
+    await mineBlocks(1, walletReceive);
+
+    const [w, tx] = await waiter;
+    assert(w);
+    assert(tx);
+  });
+
+  it('should emit `address` events', async () => {
+    const waiter = new Promise((resolve) => {
+      wdb.once('address', (w, walletKey) => resolve([w, walletKey]));
+    });
+
+    const walletReceive = await wallet.receiveAddress();
+    await mineBlocks(1, walletReceive);
+
+    const [w, walletKey] = await waiter;
+    assert(w);
+    assert(walletKey);
+  });
+
+  it('should emit `block connect` events', async () => {
+    const waiter = new Promise((resolve) => {
+      wdb.once('block connect', (entry, txs) => resolve([entry, txs]));
+    });
+
+    const walletReceive = await wallet.receiveAddress();
+    await mineBlocks(1, walletReceive);
+
+    const [entry, txs] = await waiter;
+    assert(entry);
+    assert(txs);
```

Comment:
Same comment here about testing the actual emitted values, but also I'd like to see a block with more than just a coinbase TX. Maybe even include a TX that doesnt belong to the wallet and see if it gets emitted as well (pending discussion on my top comment about filtering the TXs by wallet)

Reply:
Split the test with 2 blocks: One with a wallet tx and another without (and both are non-coinbase tx)